### ISSUE TITLE
Unable to select other entry-field by mouse-click

### ISSUE THREAD

**Issue by Sauermann** [godotengine/godot#78628]
### Godot version

v4.1.beta.custom_build [afe65de11]

### System information

Debian Linux 12 Xfce

### Issue description

For an exported `PackedVector2Array` I am unable to select a different entry field with a mouse-click after an edit.

I have verified, that reverting #76711 fixes this issue.

### Steps to reproduce

1. Load and run MRP
2. select the main node in scene tree
3. Expand the `mypacked` variable in the inspector
4. click lmb on "Add Element"
5. click lmb in the x-entry-field
6. enter number "1"
7. click lmb in the y-entry-field

After 7 I would expect that the y-entry-field is selected, but no entry field is selected.

https://github.com/godotengine/godot/assets/6299227/997515ad-4e50-48a6-9e01-d9ba90e3114d

### Minimal reproduction project

https://github.com/godotengine/godot/files/11837624/empty.zip

**Comment by akien-mga** [comment]
CC @ajreckof

**Comment by ajreckof** [comment]
Okay so I don't see how to quick fix this bug. The only way I see out of this bug is to do the proper refactoring of update property that I started working on. The refactoring wouldn't recreate editor property when it is not needed. I think I can try and make the refactor quickly so that it can be included in 4.1. But I feel like this refactor is unfortunately way too late for 4.1 as it is coming around the corner. and I'm then unsure on what we should do. I feel like this bug is on the same level of the one fixed originally. If you think the refactor would be the best way forward I can try and make it as fast as possible. :( 
- Obviously if anyone had a quickfix for this one it would solve everything and I wouldn't need to rush the refactor

**Comment by akien-mga** [comment]
I think the refactor could only happen for 4.2.

For 4.1, I would suggest reverting the original change and reopening the linked issue. I took a risk merging an EditorProperty change so close to RC, but it's good that we found the problems quickly.

What I'm not 100% of is whether just reverting #76711 is enough, or if some of the follow ups will also be invalidated then? Notably https://github.com/godotengine/godot/pull/78591 seems to fix a regression from 76711, but also another preexisting issue. I assume it's still valid even if 76711 is reverted?
